### ISSUE TITLE
Empty mocking values are not applied

### ISSUE THREAD

**Issue by adrigzr** [johanblumenberg/ts-mockito#7]
Hello! Thanks for your lib!

When trying to mock a function without an actual value, I'm encountering weird behavior. For example:

```typescript
it("should mock with empty return values", () => {
    const fn: () => number = fnmock();

    when(fn()).thenReturn(1).thenReturn();

    expect(instance(fn)()).toEqual(1);
    expect(instance(fn)()).toEqual(undefined);
    verify(fn()).called();
});
```

I expect the second mocked return method to add an `undefined` as a mocking value for the second invocation.

Is this an expected behavior?

Regards,

**Comment by johanblumenberg** [comment]
Hi @adrigzr 

Isn't it quite strange to expect the function to return `undefined`, when the function is typed to always return `number`?

I would expect the test to look something like this:

```
it("should mock with empty return values", () => {
    const fn: () => number | undefined = fnmock();

    when(fn()).thenReturn(1).thenReturn(undefined);

    expect(instance(fn)()).toEqual(1);
    expect(instance(fn)()).toEqual(undefined);
    verify(fn()).called();
});
```

If you want the mock to return `undefined` I think it is better to be explicit and call `.thenReturn(undefined)`.

**Comment by johanblumenberg** [comment]
You can pass any number of values to `.thenReturn()`, to return each value in turn. `.thenReturn(1).thenReturn(2)` is the same thing as `.thenReturn(1, 2)`.

The call `.thenReturn()` is effectively a no-op, and it is maybe a bug that it is allowed at all. The fact that `.thenResolve()` works without a value is probably a bug. I think it comes from when you have a function returning `Promise<void>` where it feels unnatural to ask the mock to return `undefined` when the type is `void`.

**Comment by johanblumenberg** [comment]
I think it might be better to change the types to not allow any of the `.thenReturn()`, `.thenResolve()`, ... methods to be invoked without any argument.

It might be an idea to allow `.thenReturn()` without a value when the retun type is void, and to allow `.thenResolve()` without a value when the return type is `Promise<void>`.

I don't think it is a good idea to allow `.thenThrow()` or `.thenReject()` without a value, because it is really not a good idea to ever throw `undefined` or `void`. You should always throw exceptions. If you anyway want to test the behaviour when something throws the value `undefined`, I think it is better to be explicit in the test and write `.thenThrow(undefined)`.

**Comment by johanblumenberg** [comment]
I created a PR to implement the suggested behaviour: https://github.com/johanblumenberg/ts-mockito/pull/9

**Comment by adrigzr** [comment]
This will do the trick for me. The main issue was the confusion when using an empty `then` method to return an `undefined` value. If this is no longer supported, it looks good to me.

**Comment by johanblumenberg** [comment]
I published version 1.0.35, so I'm closing this issue. @adrigzr Thanks for good feedback :)
